### PR TITLE
OCPBUGS-379: scan: Add conditions to scans

### PR DESCRIPTION
**Pull request by jhrozek**
Some users were confused about our re-running feature. They expected the
suites to be re-created every time and the fact that they were not and
the createTimestamp remained constant was causing questions.

Let's add conditions to the scans as well so that it's obvious that the
scan was re-ran.

Jira: OCPBUGS-379

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,9 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 - Added the ability to check default `KubeletConfig` configuration. Compliance Operator
   will fetch `KubeletConfig` for each node and save it to `/kubeletconfig/role/{{role}}`.
 
+- `ComplianceScan` custom resources now contain conditions, similar to `ComplianceSuite`
+  conditions that were implemented earlier. This aids in debugging of scheduled scans.
```
Comment (rhmdnd) [inline CHANGELOG.md:20]: Do we have a version number to use in place of earlier?

**Review by rhmdnd — COMMENTED** [review]
Looks good, one minor comment on the release note.

**Comment by mrogers950** [comment]
LGTM

File: CHANGELOG.md (hunk 1 above)
Comment (jhrozek) [reply CHANGELOG.md]: I don't know how precise my git archeology was, but I put a number into the changelog.

**Review by JAORMX — APPROVED** [review]

**Comment by JAORMX** [comment]
I think this is a nice usability improvement.

**Comment by jhrozek** [comment]
To test, create a scan (regardless of the method, can be with a ssb) and then there's two tests to do:
1. check that `.status.conditions` exists and that it contains useful information, most importantly when the scan finishes
2. it should be now possible to do e.g. `oc wait --for=condition=ready scan $scan_name`

cc @xiaojiey

**Comment by jhrozek** [comment]
I tested these FWIW:
```
$ oc wait --for=condition=processing scan upstream-ocp4-cis
compliancescan.compliance.openshift.io/upstream-ocp4-cis condition met
$ oc wait --for=condition=ready scan upstream-ocp4-cis
```

**Comment by jhrozek** [comment]
> I tested these FWIW:
> 
> ```
> $ oc wait --for=condition=processing scan upstream-ocp4-cis
> compliancescan.compliance.openshift.io/upstream-ocp4-cis condition met
> $ oc wait --for=condition=ready scan upstream-ocp4-cis
> ```

cc @Wiharris @sheriff-rh let me know if the info in this PR is enough for `docs-approved`. I can also provide the list of available conditions (it's not long) if you think it's needed, but honestly the main driver is to show that the scan had some activity and when for debugging purposes. The `oc wait --for` commands is a nice syntax sugar, but not strictly needed to document IMO.

**Comment by xiaojiey** [comment]
/label qe-approved
Details seen from https://privatebin.corp.redhat.com/?a8014970a27a5708#8ty9NfcyQ4LrZARsqVRT617VvF3gGwBc5zcmJmLdbSdb

**Comment by rhmdnd** [comment]
/retest

AWS timeout issues.

**Pull request merged**